### ISSUE TITLE
Godot 4.x script editor softlocks waiting for IBus events

### ISSUE THREAD

**Issue by adamscott** [ibus/ibus#2586]
**Which distribution and version?:** 
Fedora Linux 39 (Workstation Edition)

**Which desktop environment and version?:** 
Gnome 45.2

**Which session type?:** 
Wayland

**Which application and version?:** 
Godot 4.2

**IBus version?:** 
IBus 1.5.29-rc2

**Issue description:**
When inputting a dead key, in the Godot script editor, it softlocks. (see https://github.com/godotengine/godot/issues/85664)

`journalctl -xe` displayed this error when I typed the dead key:
```console
déc 04 14:45:03 fedora ibus-x11[71174]: ibus_input_context_post_process_key_event: Type 'h' is not supported.
```

The editor is stuck because it's waiting for multiple `XNPreeditDrawCallback` (only happens once) and `XNPreeditDoneCallback` which never happens.

[According to](https://github.com/godotengine/godot/issues/82338#issuecomment-1824710153) @HolonProduction, the issue is not reproducible when downgrading to IBus 1.5.29-rc1.

**Steps to reproduce:**
1. Open [Godot 4.2](https://godotengine.org/download/archive/4.2-stable/)
2. Create a main scene
3. Create a script to the main scene
4. Type a dead key in order to compose a new character (there's currently another issue that prevents showing the character with the underline, but it's not related. [You can try with Godot 4.0](https://godotengine.org/download/archive/4.0.4-stable/), it doesn't exhibit that particular non-relevant issue.)
5. See that the editor is stuck.

**Can you reproduce your problem when you restart ibus-daemon? (yes / no):** 
_(Run `ibus exit` and `ibus-daemon --xim &`)_

`Can not execute default panel program`

**Do you see any errors when you run ibus-daemon with the verbose option?:**
_(Run `ibus-daemon --xim --verbose &` and look at the output when you encounter your problem.)_

`Can not execute default panel program`

**Can you reproduce your problem with a new user account instead of the current your account? (yes / no):** 

Did not try yet.

**Comment by fujiwarat** [comment]
Duplicate of #2585

**Comment by SeaLiteral** [comment]
Has this bug reappeared after having been fixed for a while? Because I'm on Godot 4.3 and I still get it.
